internal/pkg/provisioner/storageclass.go

```diff
@@ -99,29 +99,33 @@ func StorageClassReservedParams(sc *storagev1.StorageClass) map[string]string {
 }
 
 // ValidateOrSetStorageClassUID returns true if the StorageClass annotation on
-// the PVC object matches the uid passed in.
+// the PVC object matches the uid passed in or the StorageClass is younger than PVC.
 //
-// If the annotation does not exist, it sets the uid passed in as the new
-// StorageClass annotation.
-func ValidateOrSetStorageClassUID(ctx context.Context, k8s client.Client, uid types.UID, obj client.Object) (bool, error) {
-	provisionedUID := obj.GetAnnotations()[StorageClassUUIDAnnotationKey]
+// If the annotation does not exist and the StorageClass is younger than PVC,
+// it sets the uid passed in as the new StorageClass annotation.
```

Comment:
Since the uid is no longer being passed in but a SC, this should be updated to `sets the uid of the StorageClass as the new...`

Reply:
Revisiting the comment above, line 104, it again says the opposite. If SC is younger than PVC, false is returned. uid isn't set.

Reply:
agreed, it should be:
```
// If the annotation does not exist and the StorageClass is older than the PVC,
// it sets the uid passed in as the new StorageClass annotation.
```

Reply:
personally I find the younger/older a bit confusing to read, but it's just me.  Maybe "was created [before|after]" would be better?

Reply:
Yes, I felt the same. Before and after sound better.